### ISSUE TITLE
bug: import/export lack support for models without genes

### ISSUE THREAD

**Issue by EuniceBabs** [SysBioChalmers/RAVEN#127]
### Error with importModel function:
I am trying to use the importModel function,

I keep getting the error below: 
``` matlab 
Index exceeds matrix dimensions.  
Error in importModel (line 850) if strcmpi(genes{1}(1:2),'G_')
 ```
What can I do?

#### Reproducing this issue:
```matlab
drosomodel=importModel('1752-0509-3-91-S3.xml')
```

#### System information
1. RAVEN version 2.0 
2. Operating system (Windows 10)

**I hereby confirm that I have:**
- [X] Followed the [guidelines](https://github.com/SysBioChalmers/RAVEN/wiki/Installation) to install RAVEN.
- [X] Checked that a similar issue does not [already exist](https://github.com/SysBioChalmers/RAVEN/issues?utf8=%E2%9C%93&q=is%3Aissue)
- [X] If suitable, needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/RAVEN) about the issue

**Comment by edkerk** [comment]
As reported on [Gitter](https://gitter.im/SysBioChalmers/RAVEN?at=5b4f409303a5fa639e63fb06), user is trying to load [this model](https://static-content.springer.com/esm/art%3A10.1186%2F1752-0509-3-91/MediaObjects/12918_2009_359_MOESM3_ESM.XML). The problem is that this model does not contain genes. This has been fixed in 13fc20b40dc8d4991d6ddefcb8e6a3d72cd750d5.

@EuniceBabs try the `fix\importModel_noGenes` [branch](https://github.com/SysBioChalmers/RAVEN/tree/fix/importModel_noGenes).

Another issue with the model is that it does not follow proper naming standards. When loading the model, RAVEN warns:
```matlab
WARNING: The following metabolite IDs are duplicates:
	4abut_c
	ac_c
	ala_B_c
	ala_L_c
	arg_L_c
	asn_L_c
	glc_D_c
	gln_L_c
	glu_L_c
	...and 8 more
```

This has nothing to do with how RAVEN deals with the file, this is just a faulty SBML file. Opening the SBML file in a text editor shows that e.g. the following L-alanine species are defined:

```xml
      <species id="M_ala_L_c" name="L-Alanine" compartment="C_c"/>
      <species id="M_ala_L_c" name="ala_L_c" compartment="C_e"/>
```
The first is in the cytosol and the id correctly ends with _c, while the second is external and its id should have ended with _e.

@EuniceBabs you can try to fix this after loading the model, with the following code:
```matlab
external=model.metComps==2;
model.mets(external)=regexprep(model.mets(external),'_[mbc]$','_e');
```
and then use `exportModel` to write a valid SBML.

**Comment by edkerk** [comment]
Also `exportModel` contained a bug when parsing models without genes. Fixed in bfea86d86c9bcab53953fcd3367aca301a201e8d

**Comment by edkerk** [comment]
@phemmy2k2 These models are stored in an older format (SBML L2V4). As the gene association is stored in _two_ locations in those models, I adapted the `importModel` code to prioritize the other location. (this is more of a work-around than a real fix, but as this is an older format, it's probably not worth the effort to fully fix this. SBML L3 models are unaffected by this). Fixed in a390f39c64a695a809990e4767ee1bf2804e600a, part of `fix/importModel_noGenes` branch.
